gazebo/common/Dem.cc

```diff
@@ -221,15 +229,40 @@ void Dem::GetGeoReference(double _x, double _y,
     OGRCoordinateTransformation *cT;
     double xGeoDeg, yGeoDeg;
 
-    // Transform the terrain's coordinate system to WGS84
-    #if GDAL_VERSION_NUM >= 2030000
-    const char *importString;
-    #else
-    char *importString;
-    #endif
-    importString = strdup(this->dataPtr->dataSet->GetProjectionRef());
-    sourceCs.importFromWkt(&importString);
-    targetCs.SetWellKnownGeogCS("WGS84");
+    // Transform the terrain's coordinate system to the appropriate
+    // coordinate system.
+    if (this->dataPtr->sphericalCoordinates->GetSurfaceType() ==
+        common::SphericalCoordinates::EARTH_WGS84)
+    {
+      #if GDAL_VERSION_NUM >= 2030000
+      const char *importString;
+      #else
+      char *importString;
+      #endif
+      importString = strdup(this->dataPtr->dataSet->GetProjectionRef());
+      sourceCs.importFromWkt(&importString);
+      targetCs.SetWellKnownGeogCS("WGS84");
+    }
+
+    else if (this->dataPtr->sphericalCoordinates->GetSurfaceType() ==
+        common::SphericalCoordinates::MOON_SCS)
+    {
+      auto importString = strdup(this->dataPtr->dataSet->GetProjectionRef());
```

Comment:
the GDAL version logic is now duplicated; let's move this outside back outside the `if` block. You can cherry-pick 58e09aa2d8b4956f3d022c43f660a9500f59ca6d if you like.

also, what should happen if the surface type is unknown? default to WGS84?

Reply:
Cheery pick done, made WGS84 as the default case in case an invalid surface type is supplied : https://github.com/osrf/gazebo/pull/3252/commits/9d1d38a3c4e99e8d5bc174c727ec6f59bda876ec